### Задача 1
Группа студентов изучает 10 различных дисциплин. Сколькими способами можно составить расписание на понедельник, если в этот день должно быть 4 разных занятия?

Для решения этой задачи используем формулу Сочетания.

In [2]:
import numpy as np

In [4]:
n = 10
k = 4
def combinations(n: int, k: int) -> int:
    return np.math.factorial(n) // (np.math.factorial(k) * np.math.factorial(n - k))
combinations(n, k)

210

Ответ: 210 способами

### Задача 2
Из колоды в 52 карты вынимают случайным образом 4 карты. Найти число исходов, соответствующих тому, что был вытянут хотя бы один туз.

Сначала с помощью фолрмулы Сочетания посчитаем количество способов вытащить 4 карты из 52

In [11]:
n = combinations(52, 4)
n

270725

Затем посчитаем количество способов вытащить 4 не туза. 52 - 4 = 48 карт бкз тузов

In [12]:
m = combinations(48, 4)
m

194580

Теперь посчитаем вероятность вытащить 4 не туза

In [13]:
P = m / n
P

0.7187367254594146

Отсюда следует, что вероятность вытащить из колоды 4 карты в числе которых будет зотя бы один туз равна:

In [14]:
P_1 = 1 - P
P_1

0.2812632745405854

### Задача 3
Семь человек рассаживаются наудачу на скамейке. Какова вероятность того, что два определённых человека будут сидеть рядом?

Воспользуемся формулой Перестановки. Сначла посчитаем кол-во способов разместить 7 человек на скамейке:

In [16]:
def permutations(n: int) -> int:
    return np.math.factorial(n)

P_7 = permutations(7)
P_7

5040

Затем посчитаем кол-во способов разместить двух человек на скамейке и оставшихся пяти человек.

In [22]:
P_2 = permutations(2)
P_5 = permutations(5)

print("P_2 =", P_2, ";", "P_5 =", P_5)

P_2 = 2 ; P_5 = 120


Всего событий, когда 2 человека из 7 сидят на скамейке рядом 6.
Посчитаем вероятность.

In [23]:
P = (6 * P_2 * P_5) / P_7
P

0.2857142857142857

Ответ: Вероятность того, что два определённых человека будут сидеть рядом равна 0.2857

### Задача 4
Из 60 вопросов, входящих в экзаменационные билеты, студент знает 50. Какова вероятность того, что среди трёх наугад выбранных вопросов студент знает: а) все? б) два?

Посчитаем кол-во способов вытянуть 3 билета из 60 и кол-во способов вытянуть 3 билет из тех, что студент знает(50) с помощью формулы Сочетания:

In [24]:
C_all = combinations(60, 3)
C_known = combinations(50, 3)
print("C_all =", C_all, ";", "C_known =", C_known)

C_all = 34220 ; C_known = 19600


Посчитаем вероятность того, что все вытянутые билеты студент знает:

In [26]:
P_all = C_known / C_all
P_all

0.5727644652250146

Теперь посчитаем кол-во способов вытащить только два известных студенту билета и кол-во способов вытащить 1 былет из незнакомых студенту:

In [28]:
С_unknown = combinations(10, 1)
C_2_known = combinations(50, 2)
print("С_unknown =", С_unknown, ";", "C_2_known =", C_2_known)

С_unknown = 10 ; C_2_known = 1225


Благоприятное число случаев, когда 2 из 3 билетов известны студенты будет равно произведению С_unknown и C_2_known. Теперь мы можем посчитать вероятность:


In [29]:
P_2_known = (С_unknown * C_2_known) / C_all
P_2_known

0.35797779076563413

Ответ: А) 0.5727644652250146, Б) 0.35797779076563413

### Задача 5

Бросается игральная кость. Пусть событие A - появление чётного числа, событие B - появление числа больше трёх. Являются ли эти события независимыми?

События A и B являются независимыми, поскольку появление четного числа, не влияет на то что это число будет больше трех, т.к. 2 - четное число меньше трех. 

примечание: если необходимо было решить эту задачу с рассчетами, буду рада увидеть решение в разборе д/з, потому что у меня ничего с этим не вышло.

### Задача 6
Допустим, имеется некоторая очень редкая болезнь (поражает 0.1% населения). Вы приходите к врачу, вам делают тест на эту болезнь, и тест оказывается положительным. Врач говорит вам, что этот тест верно выявляет 99% больных этой болезнью и всего лишь в 1% случаев даёт ложный положительный ответ.

Вопрос: какова вероятность, что вы действительно больны ей?

Подсказка: вновь используйте формулу Байеса с раскрытием знаменателя с помощью формулы полной вероятности.

Событие A - это наличие заболевание, событие B - положительный тест, тогда: 

P_A - вероятность наступления события А

P_A_ - вероятность отсутствия заболевания

P_B_A - вероятность положительного теста при наличии заболевания

P_B_A_ - вероятность положительного теста при отсутствии заболевания

In [30]:
P_A = 0.001
P_A_ = 0.999
P_B_A = 0.99
P_B_A_ = 0.01

Рассчитаем вероятность получения положительного теста преобразовав формулу полной вероятности:

In [32]:
P_B = P_B_A * P_A + P_B_A_ * P_A_
P_B

0.01098

Теперь рассчитаем вероятность наличия заболевания при положительном тесте по формуле Байеса подставив :

In [34]:
P_Bayes = (P_A * P_B_A) / P_B
P_Bayes

0.09016393442622951

Ответ: Вероятность, что вы действительно больны равна 0.09016393442622951